src/http.ts

```diff
@@ -46,8 +46,13 @@ export class HttpClient implements Client {
     protocol: "http";
 
     /** @private */
-    constructor(url: URL, authToken: string | undefined, intMode: IntMode) {
-        this.#client = hrana.openHttp(url, authToken);
+    constructor(
+        url: URL,
+        authToken: string | undefined,
+        intMode: IntMode,
+        customFetch: unknown | undefined,
```

Comment:
How about some checks on this, such as `typeof customFetch === "function"` if not undefined? Or maybe that should be pushed into the hrana client?

Also would be good to do some other runtime checks on this. For example, if it's invoked, and it fails in any way, capture errors into an error message that clearly indicates that the custom fetch function failed (so the caller can better diagnose that they provided a problematic function.

Reply:
Strictly speaking, not everything that is callable needs to have a `typedef` of `"function"`, but we can probably ignore this. What about specifying the type to be `Function | undefined`?

Yes, if the provided `fetch` function throws an exception, it will be caught and handled correctly (previously, the library only handled the case where the returned promise is rejected, but now it also handles a thrown exception).

However, I don't think that we should somehow special-case the `fetch` call: there are plenty of external functions that can fail and we can't special-case them all. The runtime provides stacktraces as a means of finding out from where an exception was thrown, and wrapping all errors from `fetch` in our own Error object would in fact make the debugging experience worse, because the original stacktrace would be lost.

Moreover, it might be very surprising if the user arranges for their `fetch` function to throw `MySpecialError`, but we intercept it and throw a different error instead.

Reply:
OK, as long as we're not muting any information that would be required for the developer to understand if their fetch function failed and how to fix it.

`Function | undefined` sounds good.